udata_hydra/crawl/select_batch.py

```diff
@@ -66,23 +65,22 @@ async def select_batch_resources_to_check() -> list[Record]:
             """
             to_check += await select_rows_based_on_query(connection, q)
 
-        # if not enough for our batch size, handle resources with outdated last check
+        # if not enough for our batch size, handle resources with planned new check
         if len(to_check) < config.BATCH_SIZE:
-            since = parse_timespan(config.SINCE)  # in seconds
-            since = datetime.now(timezone.utc) - timedelta(seconds=since)
+            now = datetime.now(timezone.utc)
             limit = config.BATCH_SIZE - len(to_check)
             q = f"""
             SELECT * FROM (
                 SELECT catalog.url, dataset_id, catalog.resource_id
                 FROM catalog, checks
                 WHERE catalog.last_check IS NOT NULL
-                AND {excluded}
                 AND catalog.last_check = checks.id
-                AND checks.created_at <= $1
+                AND checks.next_check_at >= $1
```

Comment:
It's actually nice that this logic is kept simple with the next_check_at strategy! :clap: